workflow-manager/main.go

```diff
@@ -316,6 +316,24 @@ func (b *bucket) listFiles(ctx context.Context) ([]string, error) {
 	}
 }
 
+func webIDP(sess *aws_session.Session, identity string) (*credentials.Credentials, error) {
+	arnComponents := strings.Split(identity, ":")
```

Comment:
This code isn't new, but could we use [arn.Parse](https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/#Parse) to make this a little more robust? Just think of all the test cases you won't have to write later on if you make Amazon do the parsing for you!